### PR TITLE
Define top level Permission for lint-chart.yaml (Chart Lint) workflow

### PR DESCRIPTION
**Pull request by aditya7302**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Part of #5048

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by zhzhuang-zju** [comment]
@aditya7302 thanks for your work~ Due to the path limitation https://github.com/karmada-io/karmada/blob/1255a08f5d84c2ec6c947d612642b2c35538ab99/.github/workflows/lint-chart.yaml#L16-L18 in [Chart Lint](https://github.com/karmada-io/karmada/blob/master/.github/workflows/lint-chart.yaml), you can test it on your repo by submitting a pr with modifation under the directory `charts`.

**Comment by aditya7302** [comment]
@zhzhuang-zju Here is the run of the Chart Lint workflow by making changes under the directory ```charts``` : ```https://github.com/aditya7302/karmada/actions/runs/9666757857/workflow```

File: .github/workflows/lint-chart.yaml [hunk 1]
```diff
@@ -17,6 +17,10 @@ on:
     paths:
       - "charts/**"
 
+permissions:
+  contents: read # Required by actions/checkout@v4 to fetch the repository contents.
+  statuses: write # Necessary to update the status of workflow runs.
```
Comment (zhzhuang-zju) [inline .github/workflows/lint-chart.yaml:22]:
```suggestion  
```
permission `statuses: write` is unnecessary

Comment (aditya7302) [reply .github/workflows/lint-chart.yaml:22]: @zhzhuang-zju as said by @liangyuanpeng in #5082 I have opened another PR #5126 for PR #5081. I have addressed the issue of statuses permission there.

**Pull request merged**